### PR TITLE
UI - Tag Text Search

### PR DESCRIPTION
**Pull request by johncowen**
This PR adds simple text based searching by tag.

You can now enter a tag name in the service listing pages/tabs and the result will include services with that tag.

This is a temporary text search for tags before looking into something more complex.

I've split the tests off into the first commit so if you wan to go back a commit to see it breaking you can.

### CODE REVIEW

File: ui-v2/app/controllers/dc/nodes/show.js [hunk 1]
```diff
@@ -23,6 +23,10 @@ export default Controller.extend(WithFiltering, {
       get(item, 'Service')
         .toLowerCase()
         .indexOf(term) !== -1 ||
+      (get(item, 'Tags') || [])
+        .join('')
```
[the remark this answers is not in the document]
Comment (johncowen) [reply ui-v2/app/controllers/dc/nodes/show.js]: Hmmm yeah, so I suppose with tags like 'httpd, db' a search for 'pddb' would hit, will turn this into a `some` or something 👍

[the remark this answers is not in the document]
Comment (johncowen) [reply ui-v2/app/controllers/dc/nodes/show.js]: @meirish went for a `some` for the moment at least, gonna merge but feel free to yell if something looks off with this, I added a couple more tests also, so should be good.

**Comment by cricket007** [comment]
While functional, it seems to flood the browser history by updating the url. Is that expected? 

image

**Comment by cricket007** [comment]
Might only be when there is a status that it happens. 
image

**Comment by johncowen** [comment]
Hi @cricket007 

I think there is a case to be said for either way for this (add to history/don't add to history). I actually fall down on the same side as you for this, I don't think typing into the search field should add to the history, but I'm guessing there are opinions that fall on the other side also. Happy for you to file an issue for this so we can discuss and track it, or I can do it for you? Would be nicer to get your user against a new issue though. Up to you, let me know.

Thanks,

John

**Comment by cricket007** [comment]
@johncowen I would be interested in hearing why that feature could be wanted. #4614

**Pull request merged**